### ISSUE TITLE
ZeroDivisionError when using social plugin

### ISSUE THREAD

**Issue by JefferiesTube** [squidfunk/mkdocs-material#2891]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

I think I stumbled upon another problem with the social plugin. I receive the following error / stacktrace

```
Traceback (most recent call last):
  File "c:\users\tom\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\tom\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\Tom\AppData\Local\Programs\Python\Python38\Scripts\mkdocs.exe\__main__.py", line 7, in <module>
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\click\core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\click\core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\click\core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\click\core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\mkdocs\__main__.py", line 173, in serve_command
    serve.serve(dev_addr=dev_addr, livereload=livereload, **kwargs)
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\mkdocs\commands\serve.py", line 54, in serve
    config = builder()
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\mkdocs\commands\serve.py", line 49, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\mkdocs\commands\build.py", line 284, in build
    _populate_page(file.page, config, files, dirty)
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\mkdocs\commands\build.py", line 170, in _populate_page
    page.markdown = config['plugins'].run_event(
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\mkdocs\plugins.py", line 94, in run_event
    result = method(item, **kwargs)
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\material\plugins\social\plugin.py", line 111, in on_page_markdown
    image = self.__render_card(site_name, title, description)
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\material\plugins\social\plugin.py", line 141, in __render_card
    self.__render_text((826, 328), font, title, 20),
  File "c:\users\tom\appdata\local\programs\python\python38\lib\site-packages\material\plugins\social\plugin.py", line 183, in __render_text
    if last / prev < 0.6:
ZeroDivisionError: division by zero
```

The error is gone if I disable the social plugin in the mkdocs.yml.

### Expected behaviour

Well, it crashes the build / serve. So expected is no crash / division by zero

### Actual behaviour

mkdocs crashes while building documentation

### Steps to reproduce

1. Execute `mkdocs serve`
2. Crash occurs, stack trace is logged 

### Package versions

- Python: 3.8.10
- MkDocs: 1.2.2
- Material: 7.2.1+insiders.2.12.0

### Configuration

```yaml
Mostly default stuff, the bug is gone if I disable the social plugin in the mkdocs.yml
```

### System information

- Operating system: Windows 10 
- Browser: Chrome

**Comment by squidfunk** [comment]
Thanks for reporting! Could you list the following things that are used to render the social plugin?

Values from `mkdocs.yml`, if set:

- `theme.font`

Values to render the page:

- Page title (First `h1` in document + name of the page in `mkdocs.yml` + front matter `title` if set)
- Page description (`site_description` in `mkdocs.yml` + front matter `description` if set)

For any of those values, you can just x-out all characters if you don't want to state them publicly here, but the lengths of the words are relevant, so for example:

This is some text -> XXXX XX XXXX XXXX

**Comment by JefferiesTube** [comment]
Okay, let' see.
The font part:
```
  font:
    text: Roboto
    code: Roboto Mono
   ```

The first header is `# Euro-Seriennummer` (it's again a programming exercise) the filename is `euroserialnumber.md`.

There is no `site_description` set in `mkdocs.yml` and no description on the page (it has no meta tags at all) .

**Comment by squidfunk** [comment]
I can reproduce it with "Euro-Seriennummer". If you lose the "r", it works:

![customization](https://user-images.githubusercontent.com/932156/127818231-198a1e4e-d7d9-4168-aaa9-35ac33037e61.png)


### Workaround

Workaround: Set an explicit title via Metadata, so that it breaks into more words.

``` markdown
---
title: Euro- Seriennummer
---
```

Note that this will also affect the site title. This is clearly an edge case we need to handle. At least German is full of long words, but I need a little time to think of how we could do this. We could, for example, hyphenate or set a smaller font when a single word exceeds the bounding box.

**Comment by JefferiesTube** [comment]
Would it be possible to truncate it if it exceeds the bouding box? Is so, this would be preferable instead of crashing

**Comment by squidfunk** [comment]
I agree, crashing is definitely not an option. However, note that this feature is still experimental, so bugs like those are more or less expected. I'll try to come up with a good solution next time I find the time to work on this.

**Comment by squidfunk** [comment]
The error is fixed in the latest commit. The text is now simply cut off which should be a good-enough solution for now. I'll revisit the topic (= text exceeds bounding box) when I continue working on the social plugin.

**Comment by squidfunk** [comment]
Released as part of Insiders 2.12.2.
